### PR TITLE
automate the build process (close #2) (close #1)

### LINKED ISSUE #1: make the install script idempotent
Otherwise snapshotting droplets will download them again.

### LINKED ISSUE #2: automate the build process
ref https://github.com/digitalocean/marketplace-partners

The process is outlined in https://github.com/digitalocean/marketplace-partners#getting-started

Hasura specific steps include placing https://github.com/hasura/do-marketplace/blob/master/setup-hasura.sh at https://cloudinit.readthedocs.io/en/latest/topics/modules.html#scripts-per-instance

The overall process includes
- Creating a new droplet on DigitalOcean (minimum possible conf)
- Keep the setup-hasura.sh script at the right location
- Clean up the droplet and run some checks
- Logout from droplet, shut down and snapshot
- Create a new droplet from the snapshot
- See if everything is fine
- Submit the snapshot to digitalocean

We need to automate everything until snapshotting using Packer: https://github.com/digitalocean/marketplace-partners/tree/master/packer

### PR DESCRIPTION
**Pull request by ashishra0**
## Description
Add necessary templates and scripts for packer to automate build process and restructure the repository directory structure.

Running the below packer command will start build process
`$ packer -var-file=variables.json template.json`

## Related Issues
#2

### CODE REVIEW

File: files/etc/update-mod.d/99-one-click [hunk 1]
```diff
@@ -5,27 +5,18 @@
 myip=$(hostname -I | awk '{print$1}')
 cat <<EOF
 ********************************************************************************
-
```
Comment (shahidhk) [inline files/etc/update-mod.d/99-one-click]: Some newlines are lost?

File: scripts/01-test [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+#
+# Scripts in this directory are run during the build process.
+# each script will be uploaded to /tmp on your build droplet, 
+# given execute permissions and run.  The cleanup process will
+# remove the scripts from your build system after they have run
+# if you use the build_image task.
+#
+echo "HELLO WORLD"
```
Comment (shahidhk) [inline scripts/01-test]: We don't necessarily need this test command, right?

File: scripts/99-img_check.sh [hunk 3]
```diff
@@ -4,19 +4,30 @@
 # © 2018 DigitalOcean LLC.
```
Comment (shahidhk) [inline scripts/99-img_check.sh:4]: This file is not executable. Is that a problem?

File: template.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    {
+      "type": "shell",
+      "inline": [
+        "cloud-init status --wait"
+      ]
+    },
+    {
+      "type": "shell",
+      "inline": [
+        "rm -rvf /var/lib/apt/lists/*",
```
Comment (shahidhk) [inline template.json:37]: There is an apt-get upgrade happening in the cleanup script. Let us remove that?

**Review by shahidhk — CHANGES_REQUESTED** [review]

File: scripts/01-test (hunk 2 above)
Comment (ashishra0) [reply scripts/01-test]: Nope!

File: scripts/99-img_check.sh (hunk 3 above)
Comment (ashishra0) [reply scripts/99-img_check.sh:4]: No not a problem, i did not make it executable. Should I make it executable and add it again?

File: template.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        "mv /var/lib/dpkg/status /var/lib/dpkg/status-bad",
+        "cp /var/lib/dpkg/status-old /var/lib/dpkg/status",
+        "DEBIAN_FRONTEND=noninteractive apt-get update -y",
+        "DEBIAN_FRONTEND=noninteractive apt-get dist-upgrade -y"
+      ]
+    },
+    {
+      "type": "shell",
+      "scripts": [
+        "scripts/01-test",
```
Comment (shahidhk) [inline template.json]: This will throw an error.

File: files/etc/update-mod.d/99-one-click [hunk 6]
```diff
@@ -22,10 +22,12 @@ All ports are BLOCKED except 22 (SSH), 80 (HTTP) and 443 (HTTPS).
```
Comment (shahidhk) [inline files/etc/update-mod.d/99-one-click]: File path is wrong.

**Review by shahidhk — CHANGES_REQUESTED** [review]

**Review by shahidhk — APPROVED** [review]

**Pull request merged**